src/pocketmine/Player.php

```diff
@@ -1378,7 +1385,7 @@ protected function checkNearEntities($tickDiff){
 	}
 
 	protected function processMovement($tickDiff){
-		if(!$this->isAlive() or !$this->spawned or $this->newPosition === null or $this->teleportPosition !== null or $this->isSleeping()){
```

Comment:
> Wouldn't this be considered anticheat?
It's a basic sanity check. There's no valid context in which dead players will move.